src/NuGet.Core/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs

```diff
@@ -343,5 +372,82 @@ private enum DependencyResult
             PotentiallyDowngraded,
             Cycle
         }
+
+        /// <summary>
+        /// Mark a central package version that it is transitive and need to be added to the graph.
+        /// </summary>
+        private void MarkCentralVersionForTransitiveProcessing(NuGetFramework framework, LibraryDependency libraryDependency, TransitiveCentralPackageVersions transitiveCentralPackageVersions)
+        {
+            // Set the transitive flag once is marked as transitive
+            libraryDependency.ReferenceType = LibraryDependencyReferenceType.Transitve;
+            transitiveCentralPackageVersions.TryAdd(libraryDependency);
+        }
+
+        /// <summary>
+        /// New <see cref="GraphNode{RemoteResolveResult}"/> will be created for each of the items in the <paramref name="transitiveCentralPackageVersions"/> 
+        /// and added as nodes of the <paramref name="rootNode"/>.
+        /// </summary>
+        private async Task AddTransitiveCentralPackageVersionNodesAsync(
+            GraphNode<RemoteResolveResult> rootNode,
+            LibraryDependency centralPackageVersionDependency,
+            NuGetFramework framework,
+            string runtimeIdentifier,
+            RuntimeGraph runtimeGraph,
+            TransitiveCentralPackageVersions transitiveCentralPackageVersions)
+        {
+            GraphNode<RemoteResolveResult> node = await CreateGraphNode(
+                    libraryRange: centralPackageVersionDependency.LibraryRange,
+                    framework: framework,
+                    runtimeName: runtimeIdentifier,
+                    runtimeGraph: runtimeGraph,
+                    predicate: ChainPredicate(_ => (DependencyResult.Acceptable, null), rootNode, centralPackageVersionDependency),
+                    outerEdge: null,
+                    transitiveCentralPackageVersions: transitiveCentralPackageVersions);
+
+            node.OuterNode = rootNode;
+            node.Item.CentralDependency = centralPackageVersionDependency;
+            rootNode.InnerNodes.Add(node);
+        }
+
+        /// <summary>
+        /// A centrally defined package version has the potential to become a transitive dependency.
+        /// A such dependency is defined by
+        ///     VersionCentrallyManaged = true
+        ///     ReferenceType != LibraryDependencyReferenceType.Direct
+        /// However do not include them in the graph for the begining.
+        /// </summary>
+        internal bool IsDependencyValidForGraph(LibraryDependency dependency)
+        {
+            return dependency.ReferenceType == LibraryDependencyReferenceType.Direct ||
+                   !dependency.VersionCentrallyManaged;
+        }
```

Comment:
The dependencies that need to be ignored are set with LDR.None, but I see your point.  I changed the code to be more explicit. This will not remove the need for the TransitiveCentralPackageVersions. The dependencies that needs to be processed are found at the graph traversal time.